lib/matplotlib/dates.py

```diff
@@ -212,22 +290,9 @@ def _to_ordinalf(dt):
     tzi = getattr(dt, 'tzinfo', None)
     if tzi is not None:
         dt = dt.astimezone(UTC)
-        tzi = UTC
-
-    base = float(dt.toordinal())
-
-    # If it's sufficiently datetime-like, it will have a `date()` method
-    cdate = getattr(dt, 'date', lambda: None)()
```

Comment:
Why can we delete this? Does have `np.datetime64()` this capability? I didn't find any proper docs, but a quick test says no.

Reply:
The proposed code converts `dt` from a time-zone aware datetime to the time-zone naive datetime.  As far as I can tell, that is all the old code did as well.  Is there a particular `dt` that you think will fail here?

Reply:
This seems like I could have had my own class with a `date()` method and it would have still worked. Not sure how realistic that is, but technically that would be an API change.

Reply:
This code was from #6262 @pgansle.  The tests that came in for that PR (` test_date2num_dst` and `test_date2num_dst_pandas`  still pass this PR with this code removed, so...  Maybe @pganssle has an idea of a test that would now fail?  I prefer to have a test so I can understand what the code is for....

Reply:
I don't think this is terribly important, I think this was just to distinguish between stuff like `datetime.date` and `datetime.datetime` without doing an explicit `isinstance` check. I think you can probably safely say it's an implementation detail and not part of the public interface.

The intention of that was so that if you can pass in something that duck-types as a `datetime`, it would calculate the ordinal float. I think theoretically you could add a test where you re-implement the interface for `datetime` in a separate class not derived from a `datetime` and see if you can plot that before and after this change (even if `.date()` is never called) - that would be a reasonable test of the expected public interface, but that's probably overkill.

Reply:
OK, thanks @pganssle.  The proposed change deals with `date` (as timezone naive) and `datetime`, so I think we are OK.

I agree otherwise its overkill.  If someone defined a `datetime` like object with `tzinfo` but no `astimezone` method, then I guess there could be an issue, but `tzinfo` is pretty specific thing to implement if you aren't just subclassing `datetime`.